*** pam_yubico.c Mon Jan 7 10:52:52 2013
--- pam_yubico.c Mon Jan 7 10:55:31 2013
***************
*** 111,116 ****
--- 111,118 ----
char *user_attr;
char *yubi_attr;
char *yubi_attr_prefix;
+ char *binddn;
+ char *bindpw;
int token_id_length;
enum key_mode mode;
char *chalresp_path;
***************
*** 330,337 ****
protocol = LDAP_VERSION3;
ldap_set_option (ld, LDAP_OPT_PROTOCOL_VERSION, &protocol);

! /* Bind anonymously to the LDAP server. */
! rc = ldap_simple_bind_s (ld, NULL, NULL);
if (rc != LDAP_SUCCESS)
{
DBG (("ldap_simple_bind_s: %s", ldap_err2string (rc)));
--- 332,345 ----
protocol = LDAP_VERSION3;
ldap_set_option (ld, LDAP_OPT_PROTOCOL_VERSION, &protocol);

! if (cfg->binddn && cfg->bindpw)
! {
! rc = ldap_simple_bind_s (ld, cfg->binddn, cfg->bindpw);
! }
! else{ /* Bind anonymously to the LDAP server. */
! rc = ldap_simple_bind_s (ld, NULL, NULL);
! }
!
if (rc != LDAP_SUCCESS)
{
DBG (("ldap_simple_bind_s: %s", ldap_err2string (rc)));
***************
*** 721,726 ****
--- 729,738 ----
cfg->yubi_attr = (char *) argv[i] + 10;
if (strncmp (argv[i], "yubi_attr_prefix=", 17) == 0)
cfg->yubi_attr_prefix = (char *) argv[i] + 17;
+ if (strncmp (argv[i], "binddn=", 7) == 0)
+ cfg->binddn = (char *) argv[i] + 7;
+ if (strncmp (argv[i], "bindpw=", 7) ==0)
+ cfg->bindpw = (char *) argv[i] + 7;
if (strncmp (argv[i], "token_id_length=", 16) == 0)
sscanf (argv[i], "token_id_length=%d", &cfg->token_id_length);
if (strcmp (argv[i], "mode=challenge-response") == 0)
***************
*** 751,756 ****
--- 763,770 ----
D (("user_attr=%s", cfg->user_attr ? cfg->user_attr : "(null)"));
D (("yubi_attr=%s", cfg->yubi_attr ? cfg->yubi_attr : "(null)"));
D (("yubi_attr_prefix=%s", cfg->yubi_attr_prefix ? cfg->yubi_attr_prefix : "(null)"));
+ D (("binddn=%s", cfg->binddn ? cfg->binddn : "(null)"));
+ D (("bindpw=%s", cfg->bindpw ? cfg->bindpw : "(null)"));
D (("url=%s", cfg->url ? cfg->url : "(null)"));
D (("capath=%s", cfg->capath ? cfg->capath : "(null)"));
D (("token_id_length=%d", cfg->token_id_length));

